### ISSUE TITLE
False positive with inline TypedDict and postponed annotations

### ISSUE THREAD

**Issue by object-Object** [microsoft/pyright#5935]
**Describe the bug**

When using `from __future__ import annotations` with the experimental inline TypedDict syntax, Pyright gives what I *think* is a false `reportUndefinedVariable` error for non-string forward references.

**Code or Screenshots**

Code:
```python
from __future__ import annotations
from typing import TypedDict

string: dict[{"forward_ref": "ForwardRefType"}]

# note: `value = dict[{"forward_ref": ForwardRefType}]` *does* raise NameError
non_string: dict[{"forward_ref": ForwardRefType}]  # Pyright error occurs here

class TypedDictType(TypedDict):
    forward_ref: ForwardRefType

ForwardRefType = str
```
Error:
```
(venv) PS D:\Users\redacted> python test.py
(venv) PS D:\Users\redacted> pyright test.py
D:\Users\redacted\test.py
  D:\Users\redacted\test.py:7:34 - error: "ForwardRefType" is not defined (reportUndefinedVariable)
1 error, 0 warnings, 0 informations
```

**VS Code extension or command-line**

Tested using the command line tool.
Pyright: `1.1.326`
Python: `3.11.3`

**Comment by erictraut** [comment]
Inlined TypedDict is an experimental feature intended to gather feedback about a potential future extension to the type system. It's likely to change or be removed, so please don't make use of it in any production code.

Given the status of this feature, I don't generally entertain bug fixes (especially edge cases), but this one happens to be trivial to fix, so I'll do so.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.327, which I just published. It will also be included in an upcoming release of pylance.
